lasagne/layers/autoenc.py

```diff
@@ -0,0 +1,262 @@
+        :class:`InverseLayer`. If None is provided, the layer will be linear.
+        If 'same', the :class:`InverseLayer` has the same nonlinearity as the
+        mirror :class:`Layer`.
+    b : 'new', list, Theano shared variable, numpy array, callable or None
+        An initializer for the biases. If 'new', new biases are initialized
+        using the default initialization. If a shared variable or a numpy array
+        is provided, the shape must match the incoming shape, skipping those
+        axes the biases are shared over (see below for an example). If set to
+        ``None``, the layer will have no biases and pass through its input
+        unchanged.
+        See :func:`lasagne.utils.create_param` for more information.
+
+    Returns
+    -------
+    layer: :class:`Layer` instance
+       The output :class:`Layer` of the symmetric autoencoder with
+       tied weights.
+
+    Examples
+    --------
+    >>> import lasagne
+    >>> from lasagne.layers import InputLayer, DenseLayer
+    >>> from lasagne.layers import build_autoencoder
+    >>> l_in = InputLayer((100, 20))
+    >>> l1 = DenseLayer(l_in, num_units=50)
+    >>> l2 = DenseLayer(l1, num_units=10)
+    >>> l_ae = build_autoencoder(l2, nonlinearity='same', b=None)
+    """
+
+    if isinstance(nonlinearity, (tuple, list)):
+        n_idx = 0
+
+    if isinstance(b, (tuple, list)):
+        b_idx = 0
+
+    layer = unfold_bias_and_nonlinearity_layers(layer)
+    layers = get_all_layers(layer)
+    autoencoder_layers = [layer]
+
+    for layer in layers[::-1]:
+        if not isinstance(layer, InputLayer):
+            incoming = autoencoder_layers[-1]
+
+            if isinstance(layer, NonlinearityLayer):
+                if nonlinearity == 'same':
+                    kwargs = dict(nonlinearity=layer.nonlinearity)
+                elif nonlinearity is None:
+                    kwargs = dict(nonlinearity=nonlinearities.identity)
+                elif isinstance(nonlinearity, (tuple, list)):
+                    kwargs = dict(nonlinearity=nonlinearity[n_idx])
+                    n_idx += 1
+                else:
+                    kwargs = dict(nonlinearity=nonlinearity)
+                a_layer = NonlinearityLayer(
+                    incoming=incoming,
+                    **kwargs
+                )
+            elif isinstance(layer, DropoutLayer):
+                a_layer = DropoutLayer(
+                    incoming=incoming,
+                    p=layer.p,
+                    rescale=layer.rescale
+                )
+            elif isinstance(layer, GaussianNoiseLayer):
+                a_layer = GaussianNoiseLayer(
+                    incoming=incoming,
+                    sigma=layer.sigma
+                )
+            elif isinstance(layer, BiasLayer):
+                if b is None:
+                    kwargs = dict(b=None)
+                elif b == 'new':
+                    kwargs = dict(b=init.Constant(0.))
```

Comment:
Ahaa! If that's the slightly mysterious "default bias" you mentioned above, you can just drop the `b='new'` case and have the default be `b=init.Constant(0.)` in the signature. That's a lot clearer and does exactly the same!